L1Trigger/TrackFindingTMTT/src/Histos.cc

```diff
@@ -0,0 +1,3971 @@
+
+    for (const TP& tp : vTPs) {
+      if (tp.useForAlgEff()) {
+        unsigned int nStubs = tp.numAssocStubs();  // no. of stubs in this TP.
+
+        // Number of stubs this TP has in best (eta,phi) sector, and also just dividing sectors in phi or just in eta.
+        unsigned int nStubsInBestSec = 0;
+        unsigned int nStubsInBestEtaSec = 0;
+        unsigned int nStubsInBestPhiSec = 0;
+
+        // Loop over (eta, phi) sectors.
+        for (unsigned int iPhiSec = 0; iPhiSec < numPhiSectors_; iPhiSec++) {
+          for (unsigned int iEtaReg = 0; iEtaReg < numEtaRegions_; iEtaReg++) {
+            const Sector& sector = mSectors(iPhiSec, iEtaReg);
+
+            // Count number of stubs in given tracking particle which are inside this (phi,eta) sector;
+            // or inside it if only the eta cuts are applied; or inside it if only the phi cuts are applied.
+            unsigned int nStubsInSec, nStubsInEtaSec, nStubsInPhiSec;
+            sector.numStubsInside(tp, nStubsInSec, nStubsInEtaSec, nStubsInPhiSec);
+
+            // Note best results obtained in any sector.
+            nStubsInBestSec = max(nStubsInBestSec, nStubsInSec);
+            nStubsInBestEtaSec = max(nStubsInBestEtaSec, nStubsInEtaSec);
+            nStubsInBestPhiSec = max(nStubsInBestPhiSec, nStubsInPhiSec);
+          }
+        }
+
+        // Plot fraction of stubs on each TP in its best sector.
+        hisFracStubsInSec_->Fill(float(nStubsInBestSec) / float(nStubs));
+        hisFracStubsInEtaSec_->Fill(float(nStubsInBestEtaSec) / float(nStubs));
+        hisFracStubsInPhiSec_->Fill(float(nStubsInBestPhiSec) / float(nStubs));
+      }
+    }
+
+    //=== Loop over all stubs, counting how many sectors each one appears in.
+
+    for (const Stub* stub : vStubs) {
+      // Number of (eta,phi), phi & eta sectors containing this stub.
+      unsigned int nSecs = 0;
+      unsigned int nEtaSecs = 0;
+      unsigned int nPhiSecs = 0;
+
+      // Loop over (eta, phi) sectors.
+      for (unsigned int iPhiSec = 0; iPhiSec < numPhiSectors_; iPhiSec++) {
+        for (unsigned int iEtaReg = 0; iEtaReg < numEtaRegions_; iEtaReg++) {
+          const Sector& sector = mSectors(iPhiSec, iEtaReg);
+
+          // Check if sector contains stub stub, and if so count it.
+          // Take care to just use one eta (phi) typical region when counting phi (eta) sectors.
+          if (sector.inside(stub))
+            nSecs++;
+          if (iPhiSec == 0 && sector.insideEta(stub))
+            nEtaSecs++;
+          if (iEtaReg == 0 && sector.insidePhi(stub))
+            nPhiSecs++;
+
+          // Also note which tracker layers are present in each eta sector.
+          if (iPhiSec == 0 && sector.insideEta(stub)) {
+            const TP* assocTP = stub->assocTP();
+            if (assocTP != nullptr) {
+              if (assocTP->useForAlgEff()) {
+                unsigned int lay = stub->layerId();
+                if (lay > 20)
```

Comment:
Only a dummy version of class Histos, with no implementation, will be included in the updated PR. (This will allow us to replace it will the full implementation for single-threaded L1 track development work, but will avoid the problem of histograms preventing multithreaded EDProducers for CMSSW official code)